### ISSUE TITLE
Example on querying Date field

### ISSUE THREAD

**Issue by davidhouweling** [automategreen/dynamoose#59]
Could you provide an example of how to perform a query on an established date field? This is what I have so far, but it errors out with the exception `A value provided cannot be converted into a number stack: ValidationException: A value provided cannot be converted into a number`.

Originally I tried passing just the `Date` object which didn't work, and as reading the docs go, it mentions that you convert Date to a Number, so i checked the `lib` folder and saw you were using `.getTime()` to do the conversion, but in essence `.valueOf()` is the same thing.

``` js
const date = Date.now;
return Test.query({
    date: {
      ge: date.valueOf()
    }
  }, (err, outages) => {
    if (err) {
      console.log(err);
      return;
    }
    console.log('success');
  });
```

Also, how would I be able to perform multi-condition query on the same field. For example, I would like to have a `ge` and `le` on the `date` field, would it merely just be extending the object accordingly?

**Comment by brandongoode** [comment]
`Date.now` is a function.  You need to use `Date.now()`.

You can use the `between` query.   Or you can use `.filter` to limit the results.

**Comment by davidhouweling** [comment]
Sorry, i tried to dumb it down a bit by abstracting the fact I was using `moment.js`. With the below though, it just hangs. `exec` callback never gets fired.

One thing to also note is that I have not yet put anything in the table. Do you think that could also be a problem?

``` js
  const date = moment().subtract(14, 'days');
  return Test
    .query('date')
    .between(date.valueOf(), date.clone().add(14, 'days').valueOf())
    .exec((err, result) => {
      console.log('exec callback');
      if (err) {
        console.log('errored');
        console.log(err);
        return;
      }
      console.log('success');
      console.log(result);
    });
```

**Comment by brandongoode** [comment]
That looks correct.  Can you share your Schema?

**Comment by davidhouweling** [comment]
``` js
const model = dynamoose.model('Test', {
  date: {
    type: Date,
    hashKey: true
  },
  record: {
    type: Number,
    default: 0
  }
}, { throughput: { read: 15, write: 5 } });
```

**Comment by brandongoode** [comment]
The issue is the query requires you to use the `.eq` for the hash key.  For you example, you should use `.scan`.    Sorry - I should have noticed this quicker. 

Also, the you don't need to call `.valueOf`.  You can just pass the moment objects.

**Comment by davidhouweling** [comment]
I think better error handling might be more appropriate for those scenarios, instead of it just hanging? Thanks for the assistance though.

**Comment by brandongoode** [comment]
Sorry, missed your earlier reply.

Completely agree about the error handling.  That's what took me so long to notice.  I would have expected a `QueryError` to be thrown.

```js
   if(comp !== 'EQ') {
      throw new errors.QueryError('Invalid query state; eq must follow query()');
    }
```
